src/models/depositCreditCardModel.ts

```diff
@@ -0,0 +1,78 @@
+import {action, computed, observable} from 'mobx';
+import {ROUTE_GATEWAY_FAIL, ROUTE_GATEWAY_SUCCESS} from '../constants/routes';
+import {AssetModel, WalletModel} from './';
+
+export const DEFAULT_DEPOSIT_OPTION = 'BankCard';
+
+export class DepositCreditCardModel {
+  @observable address: string = '';
+  @observable amount: number = 0;
+  @observable asset: AssetModel;
+  @observable city: string = '';
+  @observable country: string = '';
+  @observable depositOption: string = '';
+  @observable email: string = '';
+  @observable firstName: string = '';
+  @observable lastName: string = '';
+  @observable phone: string = '';
+  @observable wallet: WalletModel;
+  @observable zip: string = '';
+
+  constructor(deposit?: Partial<DepositCreditCardModel>) {
+    if (deposit) {
+      Object.assign(this, deposit);
```

Comment:
it's recommended to use `extendObservable` here since `Object.assign` will not hold mobx observables, computed and actions